### PR TITLE
docs: Explain how to combine configs

### LINKED ISSUE #17844: How to apply a legacy config to specific files
For example, applying a test config like `mocha/recommended` or `jest/recommended` only to test files seems like a fairly common practice. But I haven't seen any official documentation on this.

I found one way to do it here:

```js
module.exports = [
    // Apply mocha config only to tests.
    ...compat.extends("plugin:mocha/recommended").map(config => ({ ...config, files: ["tests/**/*.js"] })),
];
```

Another similar example:

https://github.com/psychobolt/vite-storybook-boilerplate/blob/1356effb9bdd6f8d09a8c5557e4a8aa411904c68/packages/commons/eslint.config.ts#L20-L23

It may be worth documenting an example like this in the README.

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Added a page on how to combine different types of configs into a config array. This applies to shareable configs and plugin configs...and everything in between.

fixes #17844

### CODE REVIEW

File: docs/src/use/configure/combine-configs.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+```js
+// eslint.config.js
+import exampleConfigs from "eslint-config-example";
+
+export default [
+    ...exampleConfigs.map(config => ({
+        ...config,
+        files: ["**/src/safe/*.js"]
+    }),
```
Comment (mdjermanovic) [inline docs/src/use/configure/combine-configs.md:84]:
```suggestion
    })),
```

Missing closing parenthesis of `map(`

**Review by mdjermanovic — APPROVED** [review]
LGTM, just one suggestion to fix the syntax.

File: docs/src/use/configure/combine-configs.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    parent: configure
+    title: Combine Configs
+    order: 6
+---
+
+In many cases, you won't write an ESLint config file from scratch, but rather, you'll use a combination of predefined and shareable configs along with your own overrides to create the config for your project. This page explains some of the patterns you can use to combine configs in your configuration file.
+
+## Apply a Config Object
+
+If you are importing an object from another module, in most cases, you can just insert the object directly into your config file's exported array. For example, you can use the recommended rule configurations for JavaScript by importing `recommended` config and using it in your array:
```
Comment (snitin315) [inline docs/src/use/configure/combine-configs.md:14]:
```suggestion
If you are importing an object from another module, in most cases, you can just insert the object directly into your config file's exported array. For example, you can use the recommended rule configurations for JavaScript by importing the `recommended` config and using it in your array:
```

**Review by snitin315 — APPROVED** [review]
LGTM

**Pull request merged**